Octokit/Clients/RepositoryHooksClient.cs

```diff
@@ -0,0 +1,115 @@
+        /// <param name="hookId"></param>
+        /// <returns></returns>
+        /// <remarks>See <a href="http://developer.github.com/v3/repos/hooks/#get-single-hook">API documentation</a> for more information.</remarks>
+        public Task<RepositoryHook> Get(string owner, string repositoryName, int hookId)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(owner, "owner");
+            Ensure.ArgumentNotNullOrEmptyString(repositoryName, "repositoryName");
+
+            return ApiConnection.Get<RepositoryHook>(ApiUrls.RepositoryHookById(owner, repositoryName, hookId));
+        }
+
+        /// <summary>
+        /// Creates a hook for a repository
+        /// </summary>
+        /// <remarks>See <a href="http://developer.github.com/v3/repos/hooks/#create-a-hook">API documentation</a> for more information.</remarks>
+        /// <returns></returns>
+        public Task<RepositoryHook> Create(string owner, string repositoryName, NewRepositoryHook hook)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(owner, "owner");
+            Ensure.ArgumentNotNullOrEmptyString(repositoryName, "repositoryName");
+            Ensure.ArgumentNotNull(hook, "hook");
+
+            return ApiConnection.Post<RepositoryHook>(ApiUrls.RepositoryHooks(owner, repositoryName), hook);
+        }
+
+        /// <summary>
+        /// Edits a hook for a repository
+        /// </summary>
+        /// <remarks>See <a href="http://developer.github.com/v3/repos/hooks/#edit-a-hook">API documentation</a> for more information.</remarks>
+        /// <returns></returns>
+        public Task<RepositoryHook> Edit(string owner, string repositoryName, int hookId, EditRepositoryHook hook)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(owner, "owner");
+            Ensure.ArgumentNotNullOrEmptyString(repositoryName, "repositoryName");
+            Ensure.ArgumentNotNull(hook, "hook");
+
+            return ApiConnection.Patch<RepositoryHook>(ApiUrls.RepositoryHookById(owner, repositoryName, hookId), hook);
+        }
+
+        /// <summary>
+        /// Tests a hook for a repository
+        /// </summary>
+        /// <remarks>See <a href="http://developer.github.com/v3/repos/hooks/#test-a-hook">API documentation</a> for more information. 
+        /// This will trigger the hook with the latest push to the current repository if the hook is subscribed to push events. If the hook 
+        /// is not subscribed to push events, the server will respond with 204 but no test POST will be generated.</remarks>
+        /// <returns></returns>
+        public Task Test(string owner, string repositoryName, int hookId)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(owner, "owner");
+            Ensure.ArgumentNotNullOrEmptyString(repositoryName, "repositoryName");
+
+            return ApiConnection.Post<RepositoryHookTestRequest>(ApiUrls.RepositoryHookTest(owner, repositoryName, hookId), new RepositoryHookTestRequest());
```

Comment:
Sounds like a good way to go. Especially if the pattern is already being used for one of the other http methods.

I have implemented a suggestion of the solution. 

When I am done with the rest of the review comments I will push them to the pr branch.